test/test.js

```diff
@@ -184,5 +184,51 @@ describe('observable-slim.js', _ => {
 		expect(firstProxy).to.equal(true);
 		expect(secondProxy).to.equal(true);
 	});
-	
+    it('Before change, try to add a new string property, but returning false.', () => {
+        ObservableSlim.observe(p, function (changes) {
+            expect(changes[0].type).to.equal("add");
```

Comment:
Because your `beforeChange` returns `false`, the callback function here should never execute. I would suggest setting up a test like this:

```
it('Before change, try to modify string property value, but returning false.', () => {
        callbackTriggered = false;
        ObservableSlim.observe(p, function (changes) {
            callbackTriggered = true;
        });

        ObservableSlim.beforeChange(p, function (changes) {
            return false;
        });

        test.hello = "world";
        p.hello = "WORLD";
        expect(p.hello).to.equal("world");
        expect(test.hello).to.equal("world");
        expect(callbackTriggered).to.equal(false);
    });
```